src/check.rs

```diff
@@ -45,80 +45,123 @@ enum CheckResult {
     NotFound,
 }
 
-pub fn check(chip: Chip, probe_rs_required: bool, msrv: Version) {
-    // TODO: check +nightly if needed
-    let rust_version = get_version(
-        "rustc",
-        if chip.is_xtensa() {
-            &["+esp"]
-        } else {
-            &["+stable"]
-        },
-    );
+pub fn check(chip: Chip, probe_rs_required: bool, msrv: Version, requires_nightly: bool) {
+    let rust_toolchain = if chip.is_xtensa() {
+        "esp"
+    } else if requires_nightly {
+        "nightly"
+    } else {
+        "stable"
+    };
+
+    let rust_version = get_version("rustc", &[format!("+{rust_toolchain}").as_str()]);
 
-    let rust_toolchain = if chip.is_xtensa() { "esp" } else { "stable" };
     let rust_toolchain_tool = if chip.is_xtensa() { "espup" } else { "rustup" };
 
     let espflash_version = get_version("espflash", &[]);
 
     let probers_version = get_version("probe-rs", &[]);
+
+    let esp_config_version = get_version("esp-config", &[]);
+
     let probers_suggestion_kind = if probe_rs_required {
         "required"
     } else {
         "suggested"
     };
 
-    println!("\nChecking installed versions");
+    create_check_results(
```

Comment:
You build a very large string in `create_check_results`, and now you just drop it here without printing it.